deploy/examples/cluster.yaml

```diff
@@ -161,6 +161,7 @@ spec:
 #    mgr:
 #    cleanup:
   annotations:
+#    clusterMetadata:
```

Comment:
A few suggestions around examples and docs:
- How about we move this to the end of the annotations section?
- Add a comment here with the specific configmap and secret names to which this is applied, and that `all` is not merged with it
- Perhaps we should add the full example annotation to show how these critical resources can be backed up with the annotation.
- Documentation is needed in ceph-cluster-crd.md

Reply:
Added.